src/libraries/BytesLib.sol

```diff
@@ -22,6 +22,30 @@ library BytesLib {
         }
     }
 
+    /// @notice Decode the `_arg`-th element in `_bytes` as a dynamic array
+    /// @dev The decoding of `length` and `offset` is universal,
+    /// whereas the type declaration of `res` instructs the compiler how to read it.
+    /// @param _bytes The input bytes string to slice
+    /// @param _arg The index of the argument to extract
+    /// @return length Length of the array
+    /// @return offset Pointer to the data part of the array
+    function toLengthOffset(bytes calldata _bytes, uint256 _arg)
```

Comment:
this is the same implementation in our v3 universal-router. previously removed as `CLCallDataDecoder` had this function, but now its removed. so added back v3 implementation